src/common/HipDataUtils.hpp

```diff
@@ -178,6 +177,14 @@ void deallocHipPinnedData(T& pptr)
   pptr = nullptr;
 }
 
+static std::string getHipArch()
+{  
+  hipDeviceProp_t devProp;
+  hipGetDeviceProperties(&devProp, 0);
+  std::string gcnArchName(devProp.gcnArchName);
+  std::string hipArch = gcnArchName.substr(0, 6);
```

Comment:
is 0, 6 the right thing for all architectures, aren't there 7 digit gpu names like gfx10##?

Reply:
Well, currently we're only using it for testing for gfx908 and gfx90a supported operations so it should be OK but yes more generally the full name will give you xnack information as well. An example would be like:
gfx1010:sramecc-:xnack-
So if we want to use this function for that we should grab the entire string and then sub select it based on what we're going to query.